### DISCUSSION TITLE
Why are there no flags for indicating different types of refetching?

### DISCUSSION THREAD

**Discussion by YauhenIhnatsyeu** [TanStack/query#7858] [Ideas]
Would be good to differentiate between types of refetching, for example

`const { isRefetchingOnWindowFocus, isRefetchingOnMount, isRefetchingOnCallingRefetch } = useQuery({ ... });`

**Comment by TkDodo** [comment]
because we don't currently track _why_ a refetch is made, because it usually shouldn't matter.
